library/core/src/macros/mod.rs

```diff
@@ -20,6 +20,10 @@ macro_rules! panic {
 /// Like [`assert!`], this macro has a second form, where a custom
 /// panic message can be provided.
 ///
+/// Assertions are always checked in both debug and release builds, and cannot
+/// be disabled. See [`debug_assert_eq!`] for assertions that are disabled in
+/// release builds by default.
+///
```

Comment:
I think it might flow a bit better if these were moved to after the examples. So it goes syntax discussion -> example of syntax -> additional point. Any thoughts?

I think this is a good note to add to guide anyone coming from C/Python.

Reply:
It seems to be a convention in the Rust docs that the “Examples” section (almost) always comes last.

I will move the new text earlier, so the note about syntax forms remain close to the examples, as in the existing `assert!` docs.

Perhaps we should also add headings to organize the rest of the doc text (again, like the existing `assert!` docs).